app/src/main/kotlin/com/wire/android/ui/authentication/create/email/CreateAccountEmailScreen.kt

```diff
@@ -73,29 +78,48 @@ import com.wire.android.ui.common.textfield.AutoFillTextField
 import com.wire.android.ui.common.textfield.WireTextFieldState
 import com.wire.android.ui.common.textfield.clearAutofillTree
 import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
+import com.wire.android.ui.destinations.CreatePersonalAccountOverviewScreenDestination
+import com.wire.android.ui.destinations.CreateTeamAccountOverviewScreenDestination
+import com.wire.android.ui.destinations.LoginScreenDestination
 import com.wire.android.ui.theme.wireColorScheme
 import com.wire.android.ui.theme.wireDimensions
 import com.wire.android.ui.theme.wireTypography
 import com.wire.android.util.CustomTabsHelper
 import com.wire.kalium.logic.configuration.server.ServerConfig
 
+@CreatePersonalAccountNavGraph
+@CreateTeamAccountNavGraph
+@Destination(navArgsDelegate = CreateAccountNavArgs::class)
 @Composable
-fun CreateAccountEmailScreen(viewModel: CreateAccountEmailViewModel, serverConfig: ServerConfig.Links) {
-    EmailContent(
-        state = viewModel.emailState,
-        onEmailChange = viewModel::onEmailChange,
-        onBackPressed = viewModel::goBackToPreviousStep,
-        onContinuePressed = { viewModel.onEmailContinue() },
-        onLoginPressed = viewModel::openLogin,
-        onTermsDialogDismiss = viewModel::onTermsDialogDismiss,
-        onTermsAccept = { viewModel.onTermsAccept() },
-        onErrorDismiss = viewModel::onEmailErrorDismiss,
-        tosUrl = viewModel.tosUrl(),
-        serverConfig = serverConfig
-    )
+fun CreateAccountEmailScreen(
+    createAccountEmailViewModel: CreateAccountEmailViewModel = hiltViewModel(),
+    navigator: DestinationsNavigator
+) {
+    with(createAccountEmailViewModel) {
+        EmailContent(
+            state = emailState,
+            onEmailChange = ::onEmailChange,
+            onBackPressed = { navigator.navigateUp() },
+            onContinuePressed = ::onEmailContinue,
+            onLoginPressed = {
+                navigator.navigate(LoginScreenDestination()) {
+                    if (createAccountEmailViewModel.isPersonalAccountFlow) {
+                        popUpTo(CreatePersonalAccountOverviewScreenDestination.route) { inclusive = true }
+                    } else {
+                        popUpTo(CreateTeamAccountOverviewScreenDestination.route) { inclusive = true }
+                    }
+                }
```

Comment:
If it works fine then it can stay this way, but I could suggest using our new custom `Navigator` (just like it as a param to screen like `DestinationsNavigator`) and this one can take `BackStackMode` just like our previous `NavigationManager`, so we could execute exactly the same command like it was in `viewModel.openLogin()`:
```
NavigationCommand(LoginScreenDestination(), BackStackMode.CLEAR_TILL_START)
```

Reply:
It works 🙂 I'd prefer to make the navigation on composable side than making it in viewModel